### PR TITLE
Adding lightbulb when code actions available

### PR DESCRIPTION
**Pull request by mat-mcloughlin**
Bit of a hack in this one as the promise is calling through to the subscribers for code actions.

### CODE REVIEW

**Comment by nosami** [comment]
Yes!

**Comment by mat-mcloughlin** [comment]
I can't figure out how to hook up the click event though :(

**Comment by nosami** [comment]
It can come later

**Comment by mat-mcloughlin** [comment]
Next job is. To figure out which actions work

File: lib/omnisharp-atom/features/code-action.ts [hunk 1]
```diff
@@ -43,6 +51,26 @@ class CodeAction implements OmniSharp.IFeature {
         this.disposable.add(Omni.listener.observeRuncodeaction.subscribe((data) => {
             this.applyAllChanges(data.response.Changes);
         }));
+
+        this.editor.onDidChangeCursorPosition(function (e) {
+            var oldPos = e.oldBufferPosition;
+            var newPos = e.newBufferPosition;
+
+            if (oldPos.row !== newPos.row || oldPos.column !== newPos.column) {
```
Comment (nosami) [inline lib/omnisharp-atom/features/code-action.ts]: is this 'if' needed?

Comment (mat-mcloughlin) [inline lib/omnisharp-atom/features/code-action.ts]: Possibly not

**Comment by nosami** [comment]
lightbulb

Needs debouncing

**Comment by nosami** [comment]
screen shot 2015-06-05 at 01 09 02

Also, the lightbulb sometimes multiplies :D

**Comment by nosami** [comment]
This is definitely better, but I still sometimes see multiple bulbs... also, I sometimes get the bulb for an old location after I moved the cursor.

**Comment by mat-mcloughlin** [comment]
Who the hell told you it would be ok to mess with my PR?!!!?

**Comment by mat-mcloughlin** [comment]
:P Thanks for the fixes. Tis loads better now

**Pull request merged**